> > ShapePopulationViewer.cxx, it seems to be outside the project
> 
> The project can be built in two ways:
> (1) Independent executable (or CLI) accepting command line parameters and displaying its user interface in a dedicated process
> (2) Slicer loadable module showing the user interface in Slicer and running withing the Slicer process
> 
> In https://github.com/NIRALUser/ShapePopulationViewer#build-instructions, (1) is referred as the standalone build and (2) as the Slicer extension build.
> 
> `ShapePopulationViewer.cxx` corresponds to the code responsible for implementing (1)

Hi @jcfr , thanks for the helpful info. I was trying to get (1) to work without building (2), and I figured out some of the issues. Basically there are `unset`s in the   SuperBuild/*.cmake that prevents loading existing builds. The change I made is in below.
[diff-superbuild.txt](https://github.com/slicersalt/ShapePopulationViewer/files/6354887/diff-superbuild.txt)
Meanwhile, there also is a remaining issue that prevents the final install step for ShapePopulationViewer. I paste the error info below and it seems like the install directory is not set up correctly, any suggestion? FYI, I build it on Ubuntu 20.04
`[38/41] Performing install step for 'ShapePopulationViewer'
/bin/sh: 1: : Permission denied
[38/41] Forcing configure step for 'ShapePopulationViewer'
FAILED: ShapePopulationViewer-prefix/src/ShapePopulationViewer-stamp/ShapePopulationViewer-install 
cd /home/yehan/Kitware/ShapePopulationViewer-debug/ShapePopulationViewer-build && "" "" && /home/yehan/Support/cmake-3.18.4-Linux-x86_64/bin/cmake -E touch /home/yehan/Kitware/ShapePopulationViewer-debug/ShapePopulationViewer-prefix/src/ShapePopulationViewer-stamp/ShapePopulationViewer-install
[40/41] Forcing configure step for 'ShapePopulationViewer'
ninja: build stopped: subcommand failed.
`

--- a/src/ShapePopulationQT.cxx
+++ b/src/ShapePopulationQT.cxx
@@ -206,6 +206,13 @@ void ShapePopulationQT::loadVTKFilesCLP(QFileInfoList a_fileList)
 void ShapePopulationQT::loadSRepFilesCLP(QFileInfoList a_fileList)
 {
     this->CreateWidgets(a_fileList);
+
+    //Load color map for S-Reps
+    QString filename = ":/resources/sRepColorMap.spvcm";
+    QFileInfo file(filename);
+    m_colormapDirectory= file.path();
+    gradientWidget_VISU->loadColorPointList(filename, &m_usedColorBar->colorPointList);
+    this->updateColorbar_QT();
 }
 
 void ShapePopulationQT::loadModel(vtkMRMLModelNode* modelNode)

--- a/src/ShapePopulationViewer.cxx
+++ b/src/ShapePopulationViewer.cxx
@@ -99,6 +99,30 @@ int main( int argc, char** argv )
             checkConfigurationFiles(cameraConfig, colormapConfig, shapePopulation);
         }
     }
+
+    if(!SRepFiles.empty())
+    {
+        bool load = false;
+        QFileInfoList fileList;
+        for(unsigned int i = 0 ; i < SRepFiles.size(); i++)
+        {
+            QString QFilePath(SRepFiles[i].c_str());
+            QFileInfo SRepFileInfo(QFilePath);
+            if (!QFilePath.endsWith(".xml")) wrongFileFormat(SRepFiles[i],"xml", &window);         // Control the files format
+            else if(!SRepFileInfo.exists()) fileDoesNotExist(SRepFiles[i], &window);                  // Control that the file exists
+            else
+            {
+                fileList.append(SRepFileInfo);
+                load = true;
+            }
+        }
+        if(load == true)
+        {
+            shapePopulation->loadSRepFilesCLP(fileList);
+            checkConfigurationFiles(cameraConfig, colormapConfig, shapePopulation);
+        }
+    }
+
     if(!vtkDirectory.empty())
     {
         QDir vtkDir(vtkDirectory.c_str());

--- a/src/ShapePopulationViewer.xml
+++ b/src/ShapePopulationViewer.xml
@@ -26,7 +26,7 @@ ShapePopulationViewer is a software that allows you to dynamically interact with
       <label>SRep Files</label>
       <channel>input</channel>
       <longflag>--srepfiles</longflag>
-      <flag>-v</flag>
+      <flag>-s</flag>
       <description><![CDATA[.xml Input files]]></description>
     </geometry>
 

